.github/workflows/build.yml

```diff
@@ -20,6 +18,7 @@ on:
 jobs:
   setup:
     name: Setup
+    if: ${{ github.event.label.name == 'SafeToBuild' }}
```

Comment:
Rather than putting this as if blocks within all the steps, maybe add a new top-level job of "Check Build Logic" or something that is just a "fail the build" if it's an external PR and it doesn't have the SafeToBuild label? Then the rest end up not needing to care? The failure would explain to the user and us about how it needs to be labeled by a maintainer

Reply:
I think thats definitely more elegant but given the time-pressure I'm gonna keep it simpler (for me). The "if" block only needs to be in the setup and maybe the build steps since subsequent steps are dependent via the "needs:" statements